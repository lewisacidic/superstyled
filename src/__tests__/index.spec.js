import React from "react"
import TestRenderer from "react-test-renderer"

import { lighten } from "polished"
import { color } from ".."

it("should work", () => {
  const tree = TestRenderer.create(
    <p
      css={color("red")
        .hover("green")
        .done()}
    >
      wow
    </p>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

it("should work with functions", () => {
  const tree = TestRenderer.create(
    <p
      css={color("red")
        .hover(c => "brick" + c)
        .done()}
    >
      wow
    </p>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})

it("should work with pigs", () => {
  const tree = TestRenderer.create(
    <p
      css={color("red")
        .hover(lighten(0.1))
        .done()}
    >
      wow
    </p>
  ).toJSON()

  expect(tree).toMatchSnapshot()
})
