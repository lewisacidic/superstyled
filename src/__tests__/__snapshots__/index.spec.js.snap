// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add active rules with \`.active\` 1`] = `
.emotion-0 {
  color: red;
}

.emotion-0:active {
  color: blue;
}

<button
  className="emotion-0"
>
  test content
</button>
`;

exports[`should add arbitrary rules with \`.on\` 1`] = `
.emotion-0 {
  color: red;
}

.emotion-0::before {
  color: yellow;
}

<div
  className="emotion-0"
/>
`;

exports[`should add hover rules with \`.hover\` 1`] = `
.emotion-0 {
  color: red;
}

.emotion-0:hover {
  color: green;
}

<p
  className="emotion-0"
>
  test content
</p>
`;

exports[`should add styles 1`] = `
.emotion-0 {
  color: red;
}

<p
  className="emotion-0"
>
  test content
</p>
`;

exports[`should work with functions 1`] = `
.emotion-0 {
  color: red;
}

.emotion-0:hover {
  color: brickred;
}

<p
  className="emotion-0"
>
  test content
</p>
`;
